docs/null-mapping.md

```diff
@@ -0,0 +1,125 @@
+
+Note: the examples above refer to the reader functions but similar functionality would be provided for the writer functions which would perform the reverse operation (setting the arrow null bitmap if the value is a kdb null).
+
+## Considerations
+
+The bigger problem is that unlike arrow, not all kdb types have a null value.  Also, those that do overload one value in the range (typically INT_MIN or FLOAT_MIN).  For example:
+
+- Each item in an arrow boolean array can be 0b, 1b or NULL.  Kdb has no boolean null.  Similarly it doesn't have a byte null. 
+
+- Unlike arrow, kdb can't distinguish between a null string and empty string.  Similarly it can't distinguish between the " " character and null. 
+
+- Int16 value of -32768 (a valid non-null value in arrow).  In kdb -32768 == 0Nh (null).  Similarly for the other int, float and temporal types 
+
+- Etc.
+
+Therefore mapping arrow nulls to kdb nulls is going to result in corner cases which can't be represented accurately.  However, the type mapping elsewhere in arrowkdb already has corner cases:
+
+- db has no unsigned integer types.  Arrow unsigned ints are represented in kdb as signed ints.  Therefore if the top bit of an unsigned is set in arrow, it will display as a negative number in kdb. 
+
+- Converting from kdb temporals to arrow temporals can result in a loss of precision.
+
+A compromise would be to allow the user to specify how arrowkdb should map nulls.  Each reader and writer function in arrowkdb takes an options dictionary.  A new `NULL_MAPPING option would be added which allows the user to specify whether an arrow datatype should be null mapped and what value to use for null.  For example:
+
+```q
+q)options`NULL_MAPPING 
+int16            | 0Nh 
+int32            | 0Ni 
+int64            | 0N 
+float32          | 0Ne 
+float64          | 0n 
+date32           | 0Nd 
+date64           | 0Np 
+month_interval   | 0Nm 
+day_time_interval| 0Nn 
+timestamp        | 0Np 
+time32           | 0Nt 
+time64           | 0Nn 
+duration         | 0Nn 
+utf8             | "" 
+binary           | `byte$() 
+```
+
+The type of each value in this dictionary must be the atomic type of the corresponding list representation for that datatype.  Where a datatype isn't present in this dictionary, arrowkdb would ignore the null bitmap (as per the existing behaviour).
+
+Note: There is no null mapping for arrow arrays which use nested datatypes (list, map, struct, union, dictionaries) where the parent array contains null values.  For example, an array with a struct datatype in arrow can have either null child field values or the parent struct value could be null.  Arrowkdb will only map nulls for the child fields using the above mapping.
+
+Conclusions
+
+Mapping arrow nulls to kdb nulls is considerably easier to implement and more intuitive for a kdb user.
```

Comment:
than what?

Reply:
We may add words that if user needs to verify such mapping, there is another feature - null bitmap, displaying the nulls corresponding to each field. And in my experience these two features work best in combination.